src/api/app/controllers/webui/staging/projects_controller.rb

```diff
@@ -10,16 +10,27 @@ class ProjectsController < WebuiController
 
       def create
         authorize @staging_workflow
-        staging_project = @staging_workflow.staging_projects.build(name: params[:staging_project_name])
+
+        staging_project = ::Staging::StagingProject.unscoped
+                                                   .where.not(id: @staging_workflow.project_id)
+                                                   .find_or_initialize_by(name: params[:staging_project_name])
         authorize(staging_project, :create?)
 
+        redirect_to edit_staging_workflow_path(@staging_workflow)
+
+        if staging_project.staging_workflow_id?
```

Comment:
why do you need to check the id instead of using the relationship?

`if staging_project.staging_workflow`

Reply:
I only want to know if there's one, so I avoid fetching it if there's one by checking the ID instead of using the relationship.

Reply:
taking into account that we don't have a validation, that the `id` is set doesn't mean that the object exist, but we should have a validation anyway...

Reply:
taking into account that we don't have a validation, that the `id` is set doesn't mean that the object exist

how can this be set if in the query we have ` .where.not(id: @staging_workflow.project_id)`?